sqlx-core/src/io/tls.rs

```diff
@@ -14,15 +13,24 @@ pub struct MaybeTlsStream {
 
 enum Inner {
     NotTls(TcpStream),
+    #[cfg(all(feature = "postgres", unix))]
+    UnixStream(crate::runtime::UnixStream),
```

Comment:
I don't like how this in `MaybeTlsStream`. I think it conflates what the type here is doing. 

I'd rather something like:

 * A separate `Socket` (name?) type that is internally an enumeration between `UnixStream` and `TcpStream`.

 * This `MaybeTlsStream` is then generalized around an `S` type parameter.

Thoughts?

Reply:
I don't see much reason to parameterize `MaybeTlsStream` if the type is always going to be `Socket`.

Reply:
We could call it `MaybeUdsStream` and reverse the relationship so that `MaybeUdsStream` is an enumeration between `UnixStream` and `MaybeTlsStream`.

Reply:
I guess before we get ahead of ourselves here.. is TLS over UDS even a reasonable thing? If it is, we probably need to at least support that with the type sandwhich.

Reply:
The wording in the documentation doesn't suggest TLS support at all over the domain socket: https://www.postgresql.org/docs/12/ssl-tcp.html

It doesn't really make sense anyways to encrypt over the UDS because if someone can sniff the traffic on that socket then they can probably also just inspect your process' memory or Postgres' files for juicy secrets (since they'd have to be running on the same machine already).